template/src/store.ts

```diff
@@ -0,0 +1,8 @@
+import { configureStore } from '@reduxjs/toolkit';
+import counterReducer from './features/counter/counterSlice';
+
+export default configureStore({
```

Comment:
Change this to a named export, then infer and `export type RootState = ReturnType<typeof store.getState>`.

Reply:
I'll change this but as a side question: Do you ever find this ends up in circular dependencies? I guess none of the "slice" files can use `RootState` or you'd get a circular dependency or circular type definition.

Reply:
Actually, the types _can_ be circular that way, because it's all compile-time and not runtime.  (I think "circular types" are still a thing that can be a problem, but yeah, you can round-trip `slice state -> root state -> use root state in slice`.)